pkg/controller/nodeipam/ipam/range_allocator.go

```diff
@@ -62,12 +62,15 @@ type rangeAllocator struct {
 	nodesInProcessing sets.String
 }
 
+var _ CIDRAllocator = (*rangeAllocator)(nil)
+
 // NewCIDRRangeAllocator returns a CIDRAllocator to allocate CIDRs for node (one from each of clusterCIDRs)
 // Caller must ensure subNetMaskSize is not less than cluster CIDR mask size.
 // Caller must always pass in a list of existing nodes so the new allocator.
 // Caller must ensure that ClusterCIDRs are semantically correct e.g (1 for non DualStack, 2 for DualStack etc..)
 // can initialize its CIDR map. NodeList is only nil in testing.
-func NewCIDRRangeAllocator(logger klog.Logger, client clientset.Interface, nodeInformer informers.NodeInformer, allocatorParams CIDRAllocatorParams, nodeList *v1.NodeList) (CIDRAllocator, error) {
+func NewCIDRRangeAllocator(ctx context.Context, client clientset.Interface, nodeInformer informers.NodeInformer, allocatorParams CIDRAllocatorParams, nodeList *v1.NodeList) (CIDRAllocator, error) {
```

Comment:
I'm not sure here: why is passing a context better than passing a logger?

Reply:
After reconsidering, I realized there's no clear advantage in using a context over a logger, so I've switched back to passing a logger. Thank you for pointing it out.